internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -96,12 +100,15 @@ func NewServerSidePatchHelper(original, modified client.Object, c client.Client,
 	case isNil(original):
 		hasChanges, hasSpecChanges = true, true
 	default:
-		hasChanges, hasSpecChanges = dryRunPatch(&dryRunInput{
-			path:     contract.Path{},
-			fieldsV1: getTopologyManagedFields(original),
-			original: originalUnstructured.Object,
-			modified: modifiedUnstructured.Object,
-		})
+		var err error
+		hasChanges, hasSpecChanges, err = dryRunSSAPatch(ctx, c,
+			originalUnstructured,
+			modifiedUnstructured,
```

Comment:
Can we also use an input struct and handover `dryRunUnstructured: modifiedUnstructured.DeepCopy`

Just to make 100% clear that modifiedUnstructured cannot be modified inside of this func?

Reply:
Its now kind of mixed. Passing `ctx` in a struct results in a linting error. Because of that I excluded ctx from the struct.